kcidb/db/argparse.py

```diff
@@ -0,0 +1,192 @@
+              "virtual databases.\n"
+              "You can specify a particular driver to use, and its "
+              "parameters, using the\n"
+              "-d/--database option.\n"
+              "\n"
+              "The format of the option value is <DRIVER>[:<PARAMS>], "
+              "where <DRIVER> is the\n"
+              "name of the driver, and <PARAMS> is a (sometimes optional) "
+              "driver-specific\n"
+              "parameter string.\n"
+              "\n"
+              "For example, \"-d bigquery:kernelci-production.kcidb_01\" "
+              "requests the use of\n"
+              "the \"bigquery\" database driver with the parameter string\n"
+              "\"kernelci-production.kcidb_01\", from which the driver "
+              "extracts the Google\n"
+              "Cloud project \"kernelci-production\" and the dataset "
+              "\"kcidb_01\" to connect to.\n"
+              "\n"
+              "Available drivers and format of their parameter strings "
+              "follow.\n")
+        for name, driver in parser.driver_types.items():
+            print(f"\n{name!r} driver\n" +
+                  "-" * (len(name) + 9) + "\n" +
+                  driver.get_doc())
+        parser.exit()
+
+
+def add_args(parser, database=None):
+    """
+    Add common database arguments to an argument parser.
+    Args:
+        parser:     The parser to add arguments to.
+        database:   The default database specification to use.
+    """
+    assert database is None or isinstance(database, str)
+    parser.add_argument(
+        '-d', '--database',
+        help=("Specify DATABASE to use, formatted as <DRIVER>:<PARAMS>. " +
+              "Use --database-help for more details." +
+              ("" if database is None
+               else f"Default is {database!r}.")),
+        default=database,
+        required=(database is None)
+    )
+    parser.add_argument(
+        '--database-help',
+        action=DBHelpAction,
+        help='Print documentation on database specification strings and exit.'
+    )
+
+
+class ArgumentParser(kcidb.argparse.ArgumentParser):
+    """
+    Command-line argument parser with common database arguments added.
+    """
+
+    def __init__(self, driver_types, *args, database=None, **kwargs):
+        """
+        Initialize the parser, adding common database arguments.
+
+        Args:
+            args:       Positional arguments to initialize ArgumentParser
+                        with.
+            database:   The default database specification to use, or None to
+                        make database specification required.
+            kwargs:     Keyword arguments to initialize ArgumentParser with.
+        """
+        self.driver_types = driver_types
+        super().__init__(*args, **kwargs)
+        add_args(self, database=database)
+
+
+class OutputArgumentParser(kcidb.argparse.OutputArgumentParser):
```

Comment:
We need **all** database argument parsers to have the `driver_types` attribute, implemented similar to the `ArgumentParser` above. Otherwise the `--database-help` option usage would fail for commands using them. Could you do that?